addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3915,50 +3915,66 @@ registry.BackgroundShape = SnippetOptionWidget.extend({
      * @param {function} computeShapeData function to compute the new shape data.
      */
     _handlePreviewState(previewMode, computeShapeData) {
+        const target = this.$target[0];
+        const insertShapeContainer = newContainer => {
+            const shapeContainer = target.querySelector(':scope > .o_we_shape');
+            if (shapeContainer) {
+                shapeContainer.remove();
+            }
+            if (newContainer) {
+                const preShapeLayerElement = this._getLastPreShapeLayerElement();
+                if (preShapeLayerElement) {
+                    preShapeLayerElement.after(newContainer);
```

Comment:
Pretty sure this should crash?

Reply:
Why should this crash? `childNode.after` is in the same spec as `prepend` and `replaceWith` and has [widespread support](https://caniuse.com/dom-manip-convenience). I can use jQuery's equivalent instead if you prefer though.

Reply:
Same as prepend, not a fan of using features marked as "experimental". The fact that it still experimental is a mystery... but still :shrug: